src/main/java/org/embulk/input/aws_cost_explorer/client/AwsCostExplorerClient.java

```diff
@@ -0,0 +1,88 @@
+package org.embulk.input.aws_cost_explorer.client;
+
+import com.amazonaws.services.costexplorer.AWSCostExplorer;
+import com.amazonaws.services.costexplorer.model.GetCostAndUsageRequest;
+import com.amazonaws.services.costexplorer.model.GetCostAndUsageResult;
+
+import java.util.Spliterator;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public class AwsCostExplorerClient
+{
+    private final AWSCostExplorer client;
+
+    public AwsCostExplorerClient(AWSCostExplorer client)
+    {
+        this.client = client;
+    }
+
+    /**
+     * Request all pages of the cost and usage data.
+     *
+     * @param requestParameters The request parameters for the cost and usage data.
+     * @return A stream of the cost and usage data.
+     */
+    public Stream<AwsCostExplorerResponse> requestAll(GetCostAndUsageRequest requestParameters)
+    {
+        return StreamSupport.stream(makeResponseIterator(requestParameters), false);
+    }
+
+    private Spliterator<AwsCostExplorerResponse> makeResponseIterator(GetCostAndUsageRequest requestParameters)
+    {
+        return new Spliterator<AwsCostExplorerResponse>()
+        {
+            boolean isFinished = false;
+
+            @Override
+            public boolean tryAdvance(Consumer<? super AwsCostExplorerResponse> action)
+            {
+                if (isFinished) {
+                    return false;
+                }
+
+                final AwsCostExplorerResponse response = request(requestParameters);
+                action.accept(response);
+
+                prepareNextRequest(response);
+
+                return !isFinished;
```

Comment:
nits: The isFinished variable may be deleted. (If it's an idiom, ignore it.)

```suggestion
                final AwsCostExplorerResponse response = request(requestParameters);
                action.accept(response);

                final String nextPageToken = response.getNextPageToken();
                if (nextPageToken == null || nextPageToken.isEmpty()) {
                    return false;
                }
                requestParameters.setNextPageToken(nextPageToken);

                return true;
```